src/main/java/org/aerogear/kryptowire/KryptowireServiceImpl.java

```diff
@@ -19,15 +22,16 @@
     private static final Logger logger = Logger.getLogger(KryptowireServiceImpl.class.getName());
 
     private String apiKey;
-
     private String apiEndpoint;
 
-    public KryptowireServiceImpl(String apiEndpoint, String apiKey) {
-        if (this.apiEndpoint != null && !this.apiEndpoint.isEmpty() && !this.apiEndpoint.endsWith("/")) {
-            this.apiEndpoint += "/";
+    public KryptowireServiceImpl(String apiEndpointParam, String apiKeyParam) {
+        this.setApiEndpoint(apiEndpointParam);
+        this.setApiKey(apiKeyParam);
```

Comment:
I thought using setter methods in a constructor was usually a bad idea because they can potentially be overridden causing unforeseen errors. Any reason you are using them here?

Reply:
I was getting some weird errors when running `mvn install`.. let me try to revert it back again